internal/mocks/repositoryMock.go

```diff
@@ -0,0 +1,119 @@
+// Code generated by mockery v2.40.1. DO NOT EDIT.
```

Comment:
How was this code generated? I don't see a .mockery.yaml file in the root of the project. It also looks like the code was not generated with the option to use .EXPECT() We should use .EXPECT() instead of .On()

Reply:
I generated via command line, and about the Expected, I couldn't find any tutorial on how to use, even in our projects we are using On(). But I'm validating that all mocked methods are being called a number of times

Reply:
Generating from the command-line is fine. A .mockery.yaml file should drive this. See
https://github.com/wexinc/ps-tag-dev/blob/677f0d9c7607d208a231be0a7fafd73462d1e99c/.mockery.yaml#L1-L25
For more info see the "packages" section of https://vektra.github.io/mockery/latest/features/#replace-types
An example of a test that uses expect can be found at
https://github.com/wexinc/ps-tag-dev/blob/677f0d9c7607d208a231be0a7fafd73462d1e99c/pkg/api-versioned/v1/companies/company_info_handler_test.go#L15-L95

Reply:
There is no difference between using EXPECT() and validating if the mock was called N times you wanted, like I did. I will create the mock.yaml, but for EXPECT I don't see any difference between what I did and using EXPECT()